### PR TITLE
Merge release/v2.33.1 release notes into master

### PR DESCRIPTION
**Pull request by serban-nicusor-toptal**
#### Submisison Checklist

- [ ] Run tests: `./runCmdStanTests.py src/test`
- [ ] Declare copyright holder and open-source license: see below

#### Summary:

Merge release/v2.33.1 release notes into master

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -24,4 +24,8 @@ Describe what you expect the output to be. Knowing the correct behavior is also
 Provide any additional information here.
 
 #### Current Version:
+<<<<<<< HEAD
```
Comment (WardBrian) [inline .github/ISSUE_TEMPLATE.md:27]: A lot of files now have these conflicts

**Review by WardBrian — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (serban-nicusor-toptal) [reply .github/ISSUE_TEMPLATE.md]: I found only 3 files and fixed the conflict.

**Comment by WardBrian** [comment]
@serban-nicusor-toptal this now looks like it is trying to merge unreleased changes into `master` (there were not 93 files changed between 2.33 and 2.33.1)

**Comment by serban-nicusor-toptal** [comment]
The only additional PR I merged was: https://github.com/stan-dev/cmdstan/pull/1197
As that was a fix for a failing test, should I try to revert this branch to this commit https://github.com/stan-dev/cmdstan/commit/b3778f6535b8beccb366fd8ae607f12e19dabba8 ? It will override develop merge conflict fixes but that should be fine for master branch, wdyt ?

**Comment by WardBrian** [comment]
Based on the commit log above, this also includes #1193, #1198, #1199, #1200 now.

Yes, I think you should make this PR so it only updates the change notes and version number and then merge it (ignoring CI)

**Comment by serban-nicusor-toptal** [comment]
Hey @WardBrian I've reverted the commits but now I think the Jenkins pipeline will fail because of #1197
Should we proceed with the merge anyway ?

**Comment by WardBrian** [comment]
Why is it checking out the develop branches or the dependencies on merges to master? I thought we had something to avoid this

**Comment by serban-nicusor-toptal** [comment]
Hey @WardBrian the [Jenkins part](https://github.com/stan-dev/cmdstan/blob/develop/Jenkinsfile#L89) seems to be [working fine](https://github.com/stan-dev/cmdstan/blob/develop/Jenkinsfile#L89), I think what's causing that `git checkout develop` is actually the [make stan-revert](https://github.com/stan-dev/cmdstan/blob/develop/Jenkinsfile#L88).

**Comment by serban-nicusor-toptal** [comment]
This looks good now, except the 1 warning which makes it UNSTABLE :( Maybe it's best to fix the warning and then merge so master will be green

**Comment by WardBrian** [comment]
Resolving those deprecation warnings is an open issue #1170

**Comment by serban-nicusor-toptal** [comment]
Uh, that's nice! We'll fix the warning itself rather than just ignore it in Jenkins. 
Then we can merge, the only problem is that master will show up as red (pipeline status) after the merge I think.

**Comment by WardBrian** [comment]
The fix would be in the next version, not this one

**Pull request merged**